design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,527 @@
+    reseller_id: String @doc(description: "Company re-seller ID.")
+    legal_address: CompanyLegalAddress @doc(description: "An object containing Company legal address data.")
+    company_admin: CompanyAdmin @doc(description: "An object containing information about Company Administrator.")
+    sales_representative: CompanySalesRepresentative @doc(description: "An object containing information about Company representative.")
+	payment_methods: [String] @doc(description: "List of payment methods awailable for a Company.")
+}
+
+type CompanyLegalAddress @doc(description: "Company legal address output data schema.") {
+    street: [String] @doc(description: "An array of strings that defines the Company's street address.")
+    city: String @doc(description: "City name.")
+    region: CompanyAddressRegion @doc(description: "An object containing region data for the Company.")
+    country_id: String @doc(description: "Country ID.")
+    country_name: String @doc(description: "Country name.")
+    postcode: String @doc(description: "ZIP/postal code.")
+    telephone: String @doc(description: "Company's phone number.")
+}
+
+type CompanyAdmin @doc(description: "Company Administrator (Customer with corresponding privileges) output data schema.") {
+    id: Int @doc(description: "Company Administrator's ID.")
+    email: String @doc(description: "Company Administrator email address.")
+    firstname: String @doc(description: "Company Administrator first name.")
+    lastname: String @doc(description: "Company Administrator last name.")
+    job_title: String @doc(description: "Company Administrator job title.")
+    gender: Int @doc(description: "Company Administrator gender.")
+}
+
+type CompanyAddressRegion @doc(description: "Output data schema for a region data in Company address.") {
+    region_id: Int @doc(description: "Unique region identifier. Required for certain countries.")
+    region: String @doc(description: "State or province name.")
+    region_code: String @doc(description: "Region code.")
+}
+
+type CompanySalesRepresentative @doc(description: "Company sales representative information output data schema.") {
+    email: String @doc(description: "Sales representative email address.")
+    firstname: String @doc(description: "Sales representative first name.")
+    lastname: String @doc(description: "Sales representative last name.")
+}
+
+type CompanyUsers @doc(description: "Output data schema for an object returned by a Company users search query.") {
+    items: [CompanyUser] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "An object that includes the pagination meta data of the search query.")
+}
+
+type CompanyUser @doc(description: "Company User (Customer assigned to Company) entity output data schema.") {
```

Comment:
Can we unify this with existing `Customer` via common interface? There could be irrelevant fields though, like address. My understanding address is defined on Company level and all users share it.

Reply:
> Can we unify this with existing Customer via common interface

I'd definitely like to see this done, otherwise there are missing edges in the graph folks might want to reach.

> There could be irrelevant fields though, like address. My understanding address is defined on Company level and all users share it.

I don't _think_ it would be entirely irrelevant. I'd be concerned if you could _only_ get the customer address, but I don't think anyone would expect the `Customer` object to return the `Company` address. I assume there will be use-cases we can't consider right now